packages/x-date-pickers-pro/src/DesktopTimeRangePicker/DesktopTimeRangePicker.tsx

```diff
@@ -0,0 +1,411 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { resolveComponentProps } from '@mui/base/utils';
+import {
+  extractValidationProps,
+  PickerViewRendererLookup,
+  TimeViewWithMeridiem,
+  useUtils,
+  resolveTimeFormat,
+} from '@mui/x-date-pickers/internals';
+import { renderMultiSectionDigitalClockTimeView } from '@mui/x-date-pickers/timeViewRenderers';
+import { rangeValueManager } from '../internals/utils/valueManagers';
+import { DesktopTimeRangePickerProps } from './DesktopTimeRangePicker.types';
+import { useTimeRangePickerDefaultizedProps } from '../TimeRangePicker/shared';
+import { MultiInputTimeRangeField } from '../MultiInputTimeRangeField';
+import { useDesktopRangePicker } from '../internals/hooks/useDesktopRangePicker';
+import { validateTimeRange } from '../internals/utils/validation/validateTimeRange';
+import { DateRange } from '../internals/models';
+import { renderDigitalClockTimeRangeView } from '../timeRangeViewRenderers';
+
+type DesktopTimeRangePickerComponent = (<TDate>(
+  props: DesktopTimeRangePickerProps<TDate> & React.RefAttributes<HTMLDivElement>,
+) => JSX.Element) & { propTypes?: any };
+
+const DesktopTimeRangePicker = React.forwardRef(function DesktopTimeRangePicker<TDate>(
+  inProps: DesktopTimeRangePickerProps<TDate>,
+  ref: React.Ref<HTMLDivElement>,
+) {
+  const utils = useUtils<TDate>();
+
+  // Props with the default values common to all date time pickers
+  const defaultizedProps = useTimeRangePickerDefaultizedProps<
+    TDate,
+    TimeViewWithMeridiem,
+    DesktopTimeRangePickerProps<TDate>
+  >(inProps, 'MuiDesktopTimeRangePicker');
+
+  // const thresholdToRenderTimeInASingleColumn =
+  //   defaultizedProps.thresholdToRenderTimeInASingleColumn ?? 24;
+  // const timeSteps = { hours: 1, minutes: 5, seconds: 5, ...defaultizedProps.timeSteps };
+  const shouldRenderTimeInASingleColumn = true; // (24 * 60) / (timeSteps.hours * timeSteps.minutes) <= thresholdToRenderTimeInASingleColumn;
```

Comment:
I'm pretty sure that the expected default would be to have the `MultiColumnDigitalClock` instead, just as with the `TimePicker` and the WIP `DateTimePicker`.
https://www.figma.com/file/hyNJYs7eHT73MLBHLyM9DL/v6-Date-and-Time-Pickers?type=design&node-id=736-178517&mode=dev

Reply:
Sure, I just did not add it yet :angel: